src/lib.rs

```diff
@@ -83,6 +91,45 @@ pub trait Body {
     {
         Trailers(self)
     }
+
+    /// Maps this body's data value to a different value.
+    fn map_data<F, B>(self, f: F) -> MapData<Self, F>
```

Comment:
Do you think we should add this to an Ext type? @carllerche @hawkw @seanmonstar maybe you both have some thoughts on this?

Reply:
My thoughts: I did originally make an `BodyExt` trait which had these methods. But to save having to import `BodyExt` and since all the types are local to this crate I went for putting the methods directly on `Body` instead. I don't feel strongly about this though.